### DISCUSSION TITLE
Migration from deprecated add-path on windows

### DISCUSSION THREAD

**Discussion by boneskull** [community/community#25485] [Actions]
<p>Hi,</p>
<p>I’m having trouble migrating from the <a href="https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/" rel="noopener nofollow ugc">deprecated <code>add-path</code></a> to the <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files" rel="noopener nofollow ugc">environment files</a> system.  Previously, our step looked like this:</p>
<pre><code class="lang-yaml">      - name: Add Growl Installer to Path (Windows)
        run: 'echo "::add-path::C:\Program Files (x86)\Growl for Windows"'
</code></pre>
<p>From the docs, it looks like it should be changed to:</p>
<pre><code class="lang-yaml">      - name: Add Growl Installer to Path (Windows)
        run: echo "C:\Program Files(x86)\Growl for Windows" &gt;&gt; $GITHUB_PATH
</code></pre>
<p>However, this fails to set the path as expected, and executables in the above folder cannot be found.</p>
<p>Is there a different way to do this on Windows?</p>

**Comment by chrispat** [comment]
<p>Given the default shell on Windows is PWSH it would probably be</p>
<pre><code class="lang-auto"> - name: Add Growl Installer to Path (Windows)
    run: echo "C:\Program Files(x86)\Growl for Windows" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8
</code></pre>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files" target="_blank" rel="noopener nofollow ugc">Workflow commands for GitHub Actions - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by boneskull** [comment]
<p>Thanks.  This <a href="https://github.com/mochajs/mocha/runs/1222606435?check_suite_focus=true" rel="noopener nofollow ugc">doesn’t seem to work</a>, as the <code>Growl</code> executable cannot be found.</p>
<p>Here’s <a href="https://github.com/mochajs/mocha/actions/runs/294166197/workflow" rel="noopener nofollow ugc">the workflow file in its entirety</a>.</p>
<p>It works fine <a href="https://github.com/mochajs/mocha/blob/8f5d6a9e642f12f98201f90015f525578d14881e/.github/workflows/windows-ci.yml#L76" rel="noopener nofollow ugc">using <code>add-path</code></a>, but not with the new environment file.</p>

**Comment by thboop** [comment, marked as answer]
<aside class="quote no-group" data-username="boneskull" data-post="3" data-topic="136265">
<div class="title">
<div class="quote-controls"></div>
 boneskull:</div>
<blockquote>
<p>doesn’t seem to work</p>
</blockquote>
</aside>
<p>Hey <a class="mention" href="/boneskull">@boneskull</a>, you seem to be setting different paths when using the <code>set-env</code> and <code>GITHUB_PATH</code>. Take a look at the revert PR  <a href="https://github.com/mochajs/mocha/commit/27d46b1faaace5ef897863dd842d46ec906d18b1#diff-9fa097232ca1ea1fed94685f19d2f26c" rel="noopener nofollow ugc">here</a>. Notice how the space between “Program Files” and “(x86)”.<br>
is changed as well.</p>
<p>Could you please try:</p>
<pre><code class="lang-auto">run: echo "C:\Program Files (x86)\Growl for Windows" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8
</code></pre>

**Comment by boneskull** [comment]
<p>oh jeez… it’s been that kind of week</p>

**Comment by boneskull** [comment]
<p>that worked, thanks for your help!</p>
